docs/user-guide/using-jwt-token-with-apiml.md

```diff
@@ -0,0 +1,202 @@
+
+**Example:**
+
+```bash
+curl -v -c - -X POST "https://localhost:10080/api/v1/auth/login" -d "{ \"username\": \"zowe\", \"password\": \"zowe\"}"
+```
+
+```http
+POST /api/v1/auth/login HTTP/1.1
+Accept: application/json, */*
+Content-Length: 40
+Content-Type: application/json
+
+{
+    "username": "zowe",
+    "password": "zowe"
+}
+
+HTTP/1.1 204
+Set-Cookie: apimlAuthenticationToken=eyJhbGciOiJSUzI1NiJ9...; Path=/; Secure; HttpOnly
+```
+
+#### Authenticated request
+
+One option <!-- One option to do what? --> is for the API client to pass the JWT token as a cookie header with the name `apimlAuthenticationToken`:
+
+**Example:**
+
+```http
+GET /api/v1/greeting HTTP/1.1
+Cookie: apimlAuthenticationToken=eyJhbGciOiJSUzI1NiJ9...
+
+HTTP/1.1 200
+...
+```
+
+The second option is to pass the JWT token in the `Authorization: Bearer` header:
+
+```http
+GET /api/v1/greeting HTTP/1.1
+Authorization: Bearer eyJhbGciOiJSUzI1NiJ9...
+
+HTTP/1.1 200
+...
+```
+:::tip
+The first option (using a cookie header) is recommended for web browsers with the attributes `Secure` and `HttpOnly`.
+Browsers store and send cookies automatically.
+Cookies are present on all requests, including those coming from DOM elements, and are compatible with web mechanisms such as CORS, SSE, or WebSockets.
+
+Cookies are more diffcult to support in non-web applications.
+Headers, such as `Authorization: Bearer`, can be used in non-web applications. Such headers, however, are difficult to use and secure in a web browser.
+The web application needs to store these headers and attach these headers to all requests where headers are required.
+
+Note that the API service needs to support both of them <!-- What is "both of them" --> so the API clients can use what makes more sense for them.
+:::
+
+
+### Validating tokens
+
+The API client does not need to validate tokens. API services must perform token validation themselves. If the API client receives a token from another source and needs to validate the JWT token, or needs to check details in the token, such as user ID expiration, then the client can use the `/auth/query` endpoint provided by the service.
+
+The response is a JSON response with the following fields:
+* `creation`
+* `expiration`
+* `userId` 
+
+These fields correspond to `iss`, `exp`, and `sub` JWT token claims. The timestamps are in ISO 8601 format.
+
+**Example:**
+
+```bash
+curl -k --cookie "apimlAuthenticationToken=eyJhbGciOiJSUzI1NiJ9..." -X GET "https://localhost:10080/api/v1/auth/query"
+```
+
+```http
+GET /api/v1/auth/query HTTP/1.1
+Connection: keep-alive
+Cookie: apimlAuthenticationToken=eyJhbGciOiJSUzI1NiJ9...
+
+HTTP/1.1 200
+Content-Type: application/json;charset=UTF-8
+
+{
+    "userId": "zowe",
+    "creation": "2019-11-29T13:39:18.000+0000",
+    "expiration": "2019-11-30T13:39:18.000+0000"
+}
+```
+
+<!-- It seems that the following section is a work in progress and is not ready for the published documentation -->
```

Comment:
I agree, I just started with the transfer of the information about the client usage to this page.